community/kernel/src/main/java/org/neo4j/kernel/impl/storemigration/StoreFile.java

```diff
@@ -298,16 +297,26 @@ private static void setStoreVersionTrailer( FileSystemAbstraction fs, File targe
                 throw new IllegalStateException( "Required file missing: " + targetStoreFileName );
             }
         }
-        try ( StoreChannel fileChannel = fs.open( targetStoreFileName, "rw" ) )
+        try ( PagedFile pagedFile = pageCache.map( targetStoreFileName, pageCache.pageSize() ) )
+        {
+            long trailerOffset =
+                    getTrailerOffset( pagedFile, versionTrailer.split( " " )[0] );
+            if ( trailerOffset != -1 )
+            {
+                writeTrailer( pagedFile, trailer, trailerOffset );
+            }
+        }
+    }
+
+    private static boolean storeExists( PageCache pageCache, File targetStoreFileName )
+    {
+        try ( PagedFile file = pageCache.map( targetStoreFileName, pageCache.pageSize() ) )
         {
-            fileSize = fileChannel.size();
-            fileChannel.position( fileChannel.size() - trailer.length );
-            fileChannel.write( ByteBuffer.wrap( trailer ) );
         }
-        catch ( IllegalArgumentException e )
+        catch ( IOException e )
         {
-            throw withMessage( e, e.getMessage() + " | " + "size:" + fileSize + ", trailer:" + trailer.length +
-                    " for " + targetStoreFileName );
+            return false;
         }
+        return true;
```

Comment:
This is only temporary. The only places where we are using the value return from `storeExists`, are about whether to create the store or not. I want to change those places to pass in the wish that the store files should be mapped with the CREATE option, and then the stores can add any required header bytes the file might be missing.